### PR TITLE
__init__.py: add flags named parameter for altering invocations

### PR DESCRIPTION
**Pull request by ehildenb**

### CODE REVIEW

**Review by nishantjr — APPROVED** [review]

**Review by nishantjr — CHANGES_REQUESTED** [review]
Instead of changing the command to include all flags,  use `command = '... "$in" $flags ...'` and `.variable('flags', flags)`. Otherwise, there may be multiple rules with the same name, but different flags.
 We should be doing the same for mode really.

For consistency, flags should be a string and not an array of parameters (unless you are thinking of properly wrapping it in quotes)

**Comment by ehildenb** [comment]
@nishantjr fixed if https://github.com/kframework/wasm-semantics/pull/81 passes.

File: __init__.py [hunk 1]
```diff
@@ -107,7 +107,7 @@ def target(self):
     def directory(self, *path):
         return os.path.join(self._directory, *path)
 
-    def tests(self, expected = None, glob = None, alias = None, default = True):
+    def tests(self, expected = None, glob = None, alias = None, default = True, flags = []):
```
Comment (nishantjr) [inline __init__.py]: Should be `flags = ''`

File: __init__.py [hunk 2]
```diff
@@ -117,15 +117,15 @@ def tests(self, expected = None, glob = None, alias = None, default = True):
...
+                            .then(self.runner_script(mode = 'run', flags = flags)) \
                             .then(self.proj.check(expected = e))
             if default: test.default()
             ret += [test]
         if alias is not None:
             ret = self.proj.alias(alias, ret)
         return ret
 
-    def proofs(self, glob = None, alias = None, default = True, expected = None):
+    def proofs(self, glob = None, alias = None, default = True, expected = None, flags = []):
```
Comment (nishantjr) [inline __init__.py]: same

File: __init__.py [hunk 3]
```diff
@@ -145,16 +145,17 @@ def proofs(self, glob = None, alias = None, default = True, expected = None):
     """ Low Level Interface """
 
     # mode: run|prove
-    def runner_script(self, mode):
+    def runner_script(self, mode, flags = []):
```
Comment (nishantjr) [inline __init__.py]: here too

**Review by nishantjr — CHANGES_REQUESTED** [review]

File: __init__.py (hunk 1 above)
Comment (ehildenb) [reply __init__.py]: fixed

**Review by nishantjr — APPROVED** [review]

**Pull request merged**